test/app/e2e/relay/server.js

```diff
@@ -0,0 +1,212 @@
+ * }}
+ */
+const getEnvVars = () => {
+  const apiKeySid = getEnvVar('API_KEY_SID');
+  const apiKeySecret = getEnvVar('API_KEY_SECRET');
+  const accountSid = getEnvVar('ACCOUNT_SID');
+  const authToken = getEnvVar('AUTH_TOKEN');
+  return { apiKeySid, apiKeySecret, accountSid, authToken };
+};
+
+/**
+ * @typedef {(
+ *   callSid: string,
+ *   callbackUrl: string,
+ * ) => Promise<import('axios').AxiosResponse>} CreateSubscription
+ */
+
+/**
+ * @typedef {(
+ *   callSid: string,
+ * ) => Promise<import('axios').AxiosResponse>} SendMessage
+ */
+
+/**
+ * @param {string} accountSid
+ * @param {string} apiKeySid
+ * @param {string} apiKeySecret
+ */
+const createTwilioUserDefinedMessageActions = (
+  accountSid,
+  apiKeySid,
+  apiKeySecret,
+) => {
+  const baseUrl = 'https://api.twilio.com/2010-04-01';
+  const extendedUrl = `${baseUrl}/Accounts/${accountSid}/Calls`;
+
+  /**
+   * @type {CreateSubscription}
+   */
+  const createSubscription = (callSid, callbackUrl) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessageSubscriptions.json`;
+
+    const params = new URLSearchParams();
+    params.append('Method', 'POST');
+    params.append('Callback', callbackUrl);
+
+    return axios.post(url, params, {
+      auth: {
+        username: apiKeySid,
+        password: apiKeySecret,
+      },
+    });
+  };
+
+  /**
+   * @type {SendMessage}
+   */
+  const sendMessage = (callSid) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessages.json`;
+
+    const params = new URLSearchParams();
+    params.append('Content', JSON.stringify({ message: 'ahoy, world!' }));
+
+    return axios.post(url, params, {
+      auth: {
+        username: apiKeySid,
+        password: apiKeySecret,
+      },
+    });
+  };
+
+  return { createSubscription, sendMessage };
+}
+
+/**
+ * @param {string} authToken
+ * @param {CreateSubscription} createSubscription
+ * @param {SendMessage} sendMessage
+ */
+const createExpressApp = (authToken, createSubscription, sendMessage) => {
+  const app = express();
+
+  app.use(cors());
+  app.use(express.json());
+  app.use(express.urlencoded({ extended: true }));
+  app.set('trust proxy', 1);
+
+  app.post('/create-subscription', async (req, res) => {
+    const { CallSid } = req.body;
+    if (typeof CallSid !== 'string') {
+      res.sendStatus(400);
+      throw new Error('CallSid is not of type string');
+    }
+    const callbackUrl = `${app.listener.url()}/receive-message`;
+    console.log('creating a subscription', { CallSid, callbackUrl });
+
+    const subscription = await createSubscription(CallSid, callbackUrl);
+    console.log(subscription.data);
+
+    res.json(subscription.data);
+  });
+
+  app.post('/receive-message', (req, res) => {
+    const isValidTwilioRequest = twilio.validateExpressRequest(req, authToken);
+    if (!isValidTwilioRequest) {
+      res.sendStatus(400);
+      console.warn('Incoming request is not from Twliio');
+      return;
```

Comment:
since you throw error at 149, does it make sense to throw error here as well instead of just returning?

Reply:
I opted not to throw on the first "error" since we don't want the server to crash if we receive a request from a non-Twilio source. It will simply just ignore the request and give the sender a 400. However, I opted to throw (and therefore crash) the server at the second "error" since that means Twilio sent us something wrong, and that would indicate a deeper issue in the tests.